include/natalie/gc/allocator.hpp

```diff
@@ -104,15 +101,15 @@ class Allocator {
     HeapBlock *add_heap_block() {
         auto *block = reinterpret_cast<HeapBlock *>(aligned_alloc(HEAP_BLOCK_SIZE, HEAP_BLOCK_SIZE));
         new (block) HeapBlock(m_cell_size);
-        m_blocks.push(block);
+        m_blocks.put(block, block);
         add_free_block(block);
         m_free_cells += cell_count_per_block();
         return block;
     }
 
     size_t m_cell_size;
     size_t m_free_cells { 0 };
-    Vector<HeapBlock *> m_blocks;
+    Hashmap<HeapBlock *, HeapBlock *> m_blocks;
```

Comment:
```suggestion
    Hashmap<HeapBlock *> m_blocks;
```

You don't need to specify the value type since we don't use it.